### PR TITLE
Feature/cleanup repo directory

### PR DESCRIPTION
**Pull request by evansde77**
Further cleanup from migration to pip installer. 

- Make environment lookup more flexible and better test coverage
- rename repo_directory as public method and move it to a lower level, refactor around that and update tests as needed 

@ksnavely @shudgston @appeltel @PerilousApricot

### CODE REVIEW

File: src/cirrus/environment.py [hunk 1]
```diff
@@ -10,6 +10,16 @@
 import cirrus
 import inspect
 import posixpath
+import subprocess
+
+
+def repo_directory():
```
Comment (shudgston) [inline src/cirrus/environment.py]: Docstring

File: src/cirrus/environment.py [hunk 2]
```diff
@@ -24,12 +34,21 @@ def cirrus_home():
...
     home = inspect.getsourcefile(cirrus)
-    # from the cirrus init py in the venv, we need to
-    # move up 5 dirs to get the install directory
-    for i in range(6): 
-        home = os.path.dirname(home)
+    if ('venv' in home) and ('site-packages' in home):
+        # we are in a pip installed virtualenv site-packages
+        # from the cirrus init py in the venv, we need to
+        # move up 5 dirs to get the install directory
+        for i in range(6):
```
Comment (shudgston) [inline src/cirrus/environment.py]: `for _ in range` if we don't need `i`

File: src/cirrus/environment.py [hunk 3]
```diff
@@ -24,12 +34,21 @@ def cirrus_home():
...
     
     home = inspect.getsourcefile(cirrus)
-    # from the cirrus init py in the venv, we need to
-    # move up 5 dirs to get the install directory
-    for i in range(6): 
-        home = os.path.dirname(home)
+    if ('venv' in home) and ('site-packages' in home):
+        # we are in a pip installed virtualenv site-packages
+        # from the cirrus init py in the venv, we need to
+        # move up 5 dirs to get the install directory
```
Comment (shudgston) [inline src/cirrus/environment.py]: Not clear on this. What's an example of falling into this path?
Would the cirrus import have to be imported from a pip install to include site-packages and venv in its path?

Comment (evansde77) [inline src/cirrus/environment.py]: Pip install drops it in the venv, eg: venv/lib/python2.7/site-packages 
Thinking about it, having venv in that check is dumb. 
Other path is the old installed via repo or local dev approach

File: src/cirrus/environment.py (hunk 1 above)
Comment (evansde77) [inline src/cirrus/environment.py]: fixed

File: src/cirrus/environment.py (hunk 2 above)
Comment (evansde77) [inline src/cirrus/environment.py]: fixed (and made the magic number a const with an explanatory comment)

File: src/cirrus/environment.py [hunk 4]
```diff
@@ -13,7 +13,19 @@
 import subprocess
 
 
+#
+# number of subdirectories from cirrus/__init__.py
+# when installed in a venv under CIRRUS_HOME location
+NUMBER_OF_SUBDIRS=6
```
Comment (shudgston) [inline src/cirrus/environment.py]: Spaces

File: src/cirrus/environment.py (hunk 3 above)
Comment (shudgston) [inline src/cirrus/environment.py]: These are the only cirruses (cirri?) I find under my venv, for both the installed path and the local dev path.

```
./venv/bin/cirrus
./venv/lib/python2.7/site-packages/cirrus-cli.egg-link
./venv/lib/python2.7/site-packages/cirrus.egg-link
```

Just checking for these?

File: src/cirrus/environment.py [hunk 5]
```diff
@@ -10,6 +10,28 @@
 import cirrus
 import inspect
 import posixpath
+import subprocess
+
+
+#
+# number of subdirectories from cirrus/__init__.py
+# when installed in a venv under CIRRUS_HOME location
+NUMBER_OF_SUBDIRS=6
```
Comment (shudgston) [inline src/cirrus/environment.py]: Spaces

**Comment by shudgston** [comment]
Unrelated, but this line was confusing the hell out of me https://github.com/evansde77/cirrus/blob/develop/installer2.sh#L27.

If you try searching for cirrus or cirrus-cli (https://pypi.python.org/pypi?%3Aaction=search&term=cirrus&submit=search) it doesn't show up.  I was able to find it in the global list of all packages, though.

File: src/cirrus/environment.py (hunk 3 above)
Comment (evansde77) [inline src/cirrus/environment.py]: inspect gives me the location of the cirrus init py, then it just looks at the path. If it has lib/python*/site-packages in the path it has been pip installed.

**Comment by evansde77** [comment]
@shudgston there was already a cirrus lib on pypi, so I added the -cli to it to make it unique. No idea why it doesnt show up in pip search, maybe some missing metadata in the setup.py to register it.

**Comment by shudgston** [comment]
+1

**Pull request merged**